### DISCUSSION TITLE
SPeL Expressions in SqsListener annotation

### DISCUSSION THREAD

**Discussion by adispennette** [awspring/spring-cloud-aws#1064] [Q&A]
I recently started using the SqsListener annotation and I am pretty positive I have a SPeL expression working but now it does not seem to be working as I expect. I am doing the following and I am pretty sure it should work but it is failing any type of property reference that I put in the annotation. It only seems to be working if I put the hard coded value for the queue name in.
```
 @SqsListener(queueNames = ["#{'\${sqs.s3.event.queue.names}'.split(',')}"])
```
Is there something I am missing?
This is not working either: `@SqsListener("\${sqs.s3.event.queue.names}")`

**Comment by tomazfernandes** [comment]
Hey @adispennette, 

> @SqsListener("\${sqs.s3.event.queue.names}")

Why are you escaping the $ with" \$"? Can you try without the \ please?

Thanks.

**Reply by maciejwalkowiak** [reply]
I guess it is in Kotlin, that's why it is escaped.

**Comment by tomazfernandes** [comment]
@adispennette, as you can probably tell by now I'm not too familiar with Kotlin.

Would you be able to setup a minimum project that can reproduce the issue?

Thanks.

**Reply by adispennette** [reply]
Hi, sorry this fell off my radar. I will get something set up soon.
